### DISCUSSION TITLE
Large amounts of data

### DISCUSSION THREAD

**Discussion by Laurekis** [pyswmm/pyswmm#466] [Q&A] [answered]
Hi! 
I would like help, please!
How can I change this code to get the results from all the subcatchments? 

Thanks!

`import pandas as pd
from pyswmm import Simulation, Subcatchments

# Ruta del archivo INP
inp_file = '/Users/laurasolarte/Desktop/Modelo_prueba/Modelo_prueba.inp'

time_values = []
runoff_values = []
contador_values = []

with Simulation(inp_file) as sim:
    sim.step_advance(299.999999999999999999)  
    C_22_036 = Subcatchments(sim)["C_22_036"]
    contador = 0  

    for step in sim:
        current_time = sim.current_time    
        time_values.append(current_time)  
        runoff_value = C_22_036.runoff
        runoff_values.append(runoff_value)  
        contador_values.append(contador)  
        contador += 1  # Incrementar el contador

data = {'Contador': contador_values, 'Time': time_values, 'Runoff': runoff_values}
df = pd.DataFrame(data)

print(df)`

**Comment by bemcdonnell** [comment, marked as answer]
@Laurekis, I moved this discussion from the swmmio project to the pyswmm project.  There is a pseudo-clever way to quickly fetch out results for all subcatchments that uses the lower-level swmm-python API.  Here is how I would go about doing this - just note, that this is not as clean looking as pyswmm:

```python

from pyswmm import Simulation
import pyswmm.toolkitapi as tka

with Simulation(r'Example1.inp') as sim:
    llsim_handle = sim._model

    # Create ID List
    subcatch_id_list = llsim_handle.getObjectIDList(tka.ObjectType.SUBCATCH)
    print(subcatch_id_list)

    # Launch a simulation!
    for ind, step in enumerate(sim):
        print("Sim Step {}".format(ind))
        for subcatch_id in subcatch_id_list:
            print(llsim_handle.getSubcatchResult(subcatch_id, tka.SubcResults.newRunoff))
```  

Output:

```cmd
['1', '2', '3', '4', '5', '6', '7', '8']
Sim Step 0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
```

For the list of data types that you can extract using this method, check out the enum type here (`tka.SubcResults.newRunoff`)

https://github.com/pyswmm/pyswmm/blob/22aed805ec435dce93d0ccc084041e380827e01d/pyswmm/toolkitapi.py#L152-L158
